src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -296,72 +274,88 @@ class ContentSidebar extends PureComponent<Props, State> {
+            metadataEditors: editors,
+            view: this.getDefaultSidebarView(file, editors),
+        });
     };
 
     /**
-     * File fetch success callback that sets the file and view
-     * Only set file if there is data to show in the sidebar.
-     * Skills sidebar doesn't show when there is no data.
+     * Fetches file metadata editors if required
      *
      * @private
-     * @param {Object} file - Box file
      * @return {void}
      */
-    fetchFileSuccessCallback = (file: BoxItem): void => {
+    fetchMetadata(): void {
+        const { file }: State = this.state;
         const { metadataSidebarProps }: Props = this.props;
         const { getMetadata, isFeatureEnabled = true }: MetadataSidebarProps = metadataSidebarProps;
+
+        // Only fetch metadata if we think that the file may have metadata on it
+        // but currently the metadata feature is turned off. Use case of this would be a free
+        // user who doesn't have the metadata feature but is collabed on a file from a user
+        // who added metadata on the file. If the feature is enabled we always end up showing
+        // the metadata sidebar irrespective of there being any existing metadata or not.
         const canHaveMetadataSidebar = !isFeatureEnabled && SidebarUtils.canHaveMetadataSidebar(this.props);
 
         if (canHaveMetadataSidebar) {
             this.api
                 .getMetadataAPI(true)
                 .getEditors(
-                    file,
-                    ({ editors }: { editors?: Array<MetadataEditor> }) =>
-                        this.fetchMetadataSuccessCallback(file, editors),
-                    () => this.fetchMetadataSuccessCallback(file),
+                    ((file: any): BoxItem),
+                    this.fetchMetadataSuccessCallback,
+                    noop,
                     getMetadata,
                     isFeatureEnabled,
                 );
-        } else {
-            this.fetchMetadataSuccessCallback(file);
         }
+    }
+
+    /**
+     * File fetch success callback that sets the file and sidebar visibility.
+     * Also makes an optional request to fetch metadata editors.
+     *
+     * @private
+     * @param {Object} file - Box file
+     * @return {void}
+     */
+    fetchFileSuccessCallback = (file: BoxItem): void => {
+        this.setState(
+            {
+                file,
+                view: this.getDefaultSidebarView(file),
+            },
+            this.fetchMetadata,
+        );
     };
 
     /**
      * Fetches a file
      *
      * @private
-     * @param {string} id - File id
      * @param {Object|void} [fetchOptions] - Fetch options
      * @return {void}
      */
-    fetchFile(id: string, fetchOptions: FetchOptions = {}): void {
-        if (SidebarUtils.canHaveSidebar(this.props)) {
-            this.api.getFileAPI().getFile(id, this.fetchFileSuccessCallback, this.errorCallback, {
+    fetchFile(fetchOptions: FetchOptions = {}): void {
```

Comment:
This function should completely go away if/when each sidebar fetch their own file object